lib/oli_web/live/insights/insights.ex

```diff
@@ -533,6 +531,34 @@ defmodule OliWeb.Insights do
     end
   end
 
+  defp get_by_page_row(socket, :by_objective) do
+    if socket.assigns.is_product do
+      Oli.Analytics.ByObjective.query_against_project_slug(
+        socket.assigns.project.slug,
+        socket.assigns.product_ids
+      )
+    else
+      Oli.Analytics.ByObjective.query_against_project_slug(
+        socket.assigns.project.slug,
+        socket.assigns.section_ids
```

Comment:
I'd like to clarify: How do you differentiate between a product_id and a section_id when they are passed as the second argument to the query_against_project_slug/2 function? Both are integers, right?

Reply:
section_ids and product_ids are both lists of integers. Yes, there are two handle_events: section_change and product_change. One handles the list of section_ids corresponding to what the user selects on the multiselect component, and the other handles the product_ids. Additionally, there is a variable called is_product, which is a boolean indicating which list (product_ids or section_ids) has to be passed in the query.

Reply:
As we discussed previously, both products and sections utilize the same struct, which is why they can be used interchangeably for the query_against_project_slug/2 function without needing a specific distinction.